File: csharp-rest/DataModels/Symbol.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csharp_rest {
    public class Symbol {
        public string symbol_id { get; set; }
        public string exchange_id { get; set; }
        public string symbol_type { get; set; }
        public bool option_type_is_call { get; set; }
        public decimal option_strike_price { get; set; }
        public decimal option_contract_unit { get; set; }
        public string option_exercise_style { get; set; }
        public DateTime option_expiration_time { get; set; }
        public DateTime future_delivery_time { get; set; }
        public string asset_id_base { get; set; }
        public string asset_id_quote { get; set; }
    }
}

File: data-api/haskell-rest/CoinApi/Types/Symbol.hs
{-# LANGUAGE OverloadedStrings #-}

module CoinApi.Types.Symbol where

import CoinApi.Types.Internal
import Text.Read (readEither)
import Data.Aeson.Types

data SymbolType = SPOT | FUTURE | OPTION
  deriving (Show, Read, Eq)

instance FromJSON SymbolType where
  parseJSON = withText "SymbolType" $ \text -> case readEither (unpack text) of
                                                 Right sym -> return sym
                                                 Left err  -> fail err

data Symbol = Symbol { symbol_id              :: !Text
                     , exchange_id            :: !Text
                     , symbol_type            :: !SymbolType
                     , asset_id_base          :: !Text
                     , asset_id_quote         :: !Text
                     , future_delivery_time   :: !UTCTime
                     , option_type_is_call    :: !Bool
                     , option_strike_price    :: !Scientific
                     , option_contract_unit   :: !Int
                     , option_exercise_style  :: !Text
                     , option_expiration_time :: !UTCTime }
  deriving (Show, Eq)

instance FromJSON Symbol where
  parseJSON = withObject "Symbol" $ \o -> Symbol
                                          <$> o .: "symbol_id"
                                          <*> o .: "exchange_id"
                                          <*> o .: "symbol_type"
                                          <*> o .: "asset_id_base"
                                          <*> o .: "asset_id_quote"
                                          <*> fmap fromTime (o .:: "future_delivery_time")
                                          <*> o .:: "option_type_is_call"
                                          <*> o .:: "option_strike_price"
                                          <*> o .:: "option_contract_unit"
                                          <*> o .:: "option_exercise_style"
                                          <*> fmap fromTime (o .:: "option_expiration_time")
